src/mgmt/services/manager/project/api/nodes.py

```diff
@@ -62,11 +227,58 @@ def single_node(node_id):
         db.session.commit()
         response_object = node.to_json()
     elif request.method == 'DELETE':
+        post_data = request.get_json()
+        node_name = node.name
+        zgc_id = node.zgc_id
+        all_droplets_in_zgc = obj_api.list_cluster_custom_object(group='zeta.com',
+                                                                 version='v1',
+                                                                 plural='droplets',
+                                                                 label_selector='zgc_id='+zgc_id+',network=tenant'
+                                                                )['items']
+        droplet_to_remove = None
+
+        tenant_droplet_name = 'droplet-tenant-'+node_name.replace('_', "-").lower()
+        zgc_droplet_name = 'droplet-zgc-'+node_name.replace('_', "-").lower()
+
+        for droplet in all_droplets_in_zgc:
+            if droplet['metadata']['name'] == tenant_droplet_name:
+                droplet_to_remove = droplet
+                all_droplets_in_zgc.remove(droplet)
+                break
+
+        if droplet_to_remove is None:
+            logger.error('Cannot find that droplet with name: {}, thus not deleting anything'.format(tenant_droplet_name))
+            return jsonify({'error': 'Droplet not found'})
+        else:
+            ip_to_assign = droplet_to_remove['spec']['ip']
+
+            ip_amount = len(ip_to_assign)
+
+            number_of_droplets_to_assign = len(all_droplets_in_zgc)
+
+            # Sort these droplets in accending order, so the droplet with the least IPs will be in the front.
+            all_droplets_in_zgc.sort(key=lambda x : len(x['spec']['ip']), reverse=False)
+            modified_droplets = dict()
+            for index in range(ip_amount):
+                ip = ip_to_assign[index]
+                mac = get_mac_from_ip(ip)
+                all_droplets_in_zgc[index % number_of_droplets_to_assign]['spec']['ip'].append(ip)
+                all_droplets_in_zgc[index % number_of_droplets_to_assign]['spec']['mac'].append(mac)
+                modified_droplet_name = all_droplets_in_zgc[index % number_of_droplets_to_assign]['metadata']['name']
+                if modified_droplet_name not in modified_droplets: 
+                    modified_droplets[modified_droplet_name] = all_droplets_in_zgc[index % number_of_droplets_to_assign]
+
+            delete_droplet(name=tenant_droplet_name)
+            delete_droplet(name=zgc_droplet_name)
+
+            for droplet_name in modified_droplets:
+                update_droplet(modified_droplets[droplet_name])
+
         db.session.delete(node)
         db.session.commit()
         response_object = {}
     return jsonify(response_object)
 
 
 if __name__ == '__main__':
-    app.run()
+    app.run()
```

Comment:
why there are these kind of whitespace changes? don't remove newline from end of file

Reply:
fixed